### PR TITLE
OS-43 many rails version in ci

### PR DESCRIPTION
**Pull request by choosen**
I fixed also problem with spec (rails 4.2 and 6.1) by adding some dummy parameter instead of empty params

Bundle-audit is disabled for rails 4.2 @ CI: Activesupport & actionpack have some flaws there.

based on: 
https://www.schneems.com/2021/01/13/migrating-a-ruby-library-from-travisci-to-circleci/
https://circleci.com/blog/circleci-matrix-jobs/
https://circleci.com/docs/2.0/configuration-reference/#matrix-requires-version-21
https://github.com/collectiveidea/audited/blob/master/spec/spec_helper.rb

> I did not test pushing codecov on master branch

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -8,3 +8,19 @@
 def jsonapi_response
   JSON.parse(response.body, symbolize_names: true)
 end
+
+def post_with_rails_fix(url, params: {}, **others)
```
Comment (Marahin) [inline spec/rails_helper.rb:12]: Love this, but perhaps some comment that would wrap this method would be useful for future developers to come and maintain?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -8,3 +8,19 @@
...
+  if Rails::VERSION::MAJOR >= 5
+    post url, params: params, **others
+  else
+    post url, params, **others
+  end
+end
+
+def patch_with_rails_fix(url, params: {}, **others)
```
Comment (Marahin) [inline spec/rails_helper.rb:20]: Same as with `post_with_rails_fix` - some additional commented info would be great to understand why is this necessary and where does `others` come from

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,12 @@
+root = true
```
Comment (Marahin) [inline .editorconfig:1]: This file does not seem to be related to the gem and we probably should keep it out the repository, just like we do with `.idea` files.

**Review by Marahin — CHANGES_REQUESTED** [review]
Overall things look good, well done. I have some minor suggestions left in the comments.

File: .editorconfig (hunk 3 above)
Comment (choosen) [reply .editorconfig:1]: I think it's not the same.
editorconfig is setup for all editors (some require plugin for it)
Idea is only for intelliJ (?)

as a reference see:
https://github.com/ruby/ruby/blob/master/.editorconfig

It's more like rubocop setttings for development purposes.

Comment (Marahin) [reply .editorconfig:1]: Ok, that seems like pretty cool idea, haven't seen that before.

File: spec/rails_helper.rb [hunk 4]
```diff
@@ -9,27 +9,30 @@ def jsonapi_response
...
 
-# rails changed test case request params from positional to keywords in rails 5.1.7
+# rails changed controller params from positional to keywords in rails 5.1.7
 # 5.1.6>= rails >=5.0 consumes both ways.
 # https://apidock.com/rails/v5.1.7/ActionController/TestCase/Behavior/process
 # others keywords can be: session, body, flash, format, xhr, as.
 # We are using `as: json` in specs (rails 5+),
 # but rails 4 does not require providing that and consumes only params, session and flash.
+# Integration specs does not have session and flash but we don't require them so we will skip them.
+# Integration reqests can pass also headers, but we don't use it yet.
```
Comment (Marahin) [inline spec/rails_helper.rb:19]: Typo "reqests" - "requests"

Also I think it would sound better: "requests can also pass headers", instead of "can pass also headers"

**Review by Marahin — CHANGES_REQUESTED** [review]
Just a tiny suggestion on comments, other than that LGTM! :shipit:

**Review by Marahin — APPROVED** [review]
